Patch 1

abhiramtilakiiit opened the pull request:
fixes #71 

## Change Summary

- Written a Manacher's Algorithm from scratch.
- Added 2 leetcode problems and 
- Attached 3 suggested problems

## Checklist

**If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed and will be closed without notice. Regular contributors (with 10+ PRs) can skip this part.**

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [x] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [x] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Tutorial

- [x] I've read and followed the [Tutorial Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorial-template)
- [x] I've explained my topic well with 2 - 3 LC problems and no external problems are used. 
- [x] I've provided the full working solutions to the problems used in this tutorial.
- [x] I've provided suggested problems at the end with the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] I've given credits / references if I use external resources. (For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).)

### Solutions

- [ ] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [ ] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [ ] I've confirmed that comments are put above each line rather than writing on the same line.
- [ ] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

wingkwong reviewed (commented):
Overall it looks good. Just a few comments.

1. The solution in example 2 is same as the one in example 1. please fix it.
2. for the code, please add spaces before & after operators, e.g. `pLengths[2*i] % 2` -> `pLengths[2 * i] % 2`. (apply to all)
3. How to apply a few modications to make it work with even arrays?
4. I'd suggest to make a template / function so that people can easily use it.

abhiramtilakiiit commented:
Alright, I've done the following changes:
1. My bad, function name was still `longest_palindrome`, fixed it.
2. Added spaces around operators
3. I don't really understand what you mean by that, odd or even my algo inserts '#'s between each character
4. Added template, and linked to it in the tutorial

wingkwong reviewed (commented):
1. Verified.
2. I can still see there is lots of operators packed together. e.g. `len(s)-1`, `len(pLengths)/2`, `expand_index+pLengths[i]`, etc ... Please go through the code and revise them accordingly. 
3. My bad. I found it now. Ignore this comment.
4. You template should be a function. Given a string, return the longest palindromic substring. so that people can plug and use directly.
5. The link in "We've created a template for Manacher's algorithm implementation in different languages so that its easy to use." is not correct.

abhiramtilakiiit commented:
I added the template code in function, but a more general case than just 'longest palindrome'. I return set of all palindromic strings

wingkwong reviewed (approved):
I'll fine tune a bit later.

abhiramtilakiiit commented:
> I'll fine tune a bit later.

Alright, Thank You
